geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/ClientUpdateMessageImpl.java

```diff
@@ -1573,7 +1574,13 @@ public CqNameToOpHashMap(CqNameToOpSingleEntry se) {
     public void sendTo(DataOutput out) throws IOException {
       // When serialized it needs to look just as if writeObject was called on a HASH_MAP
       out.writeByte(DSCODE.HASH_MAP.toByte());
-      DataSerializer.writeHashMap(this, out);
+      // By copying the hashmap prior to serializing, we avoid dealing with atomicity issues during
+      // serialization. When we serialize a hashmap, we first write the size to the data output
+      // then each of the entires. If this map is modified between writing the size and iterating
+      // the entries, there is a possibility of a mismatch of size and number of entries written
+      // to the data output.
+      Map<String, Integer> mapToSerialize = new ConcurrentHashMap<>(this);
```

Comment:
Since the new map is not shared across threads why not just use HashMap? I suspect there is overhead in the ConcurrentHashMap we don’t need.

Reply:
Yeah you are correct - I will change it to a regular HashMap.  The important thing is that the parameter to the copy constructor (`this`) is thread-safe, which it is.  I'll change it, thanks.

Reply:
Just FYI, the reason I picked ConcurrentHashMap here initially stemmed from looking at the copy constructor logic of both HashMap and ConcurrentHashMap.  It looked like the sizing logic would potentially oversize the underlying table in a HashMap, but even if it shouldn't cause any functional issues, so I'm fine with changing this to a HashMap.